### DISCUSSION TITLE
Errors (non-fatal) in RegridDataPlane and ExtractTiles use-case processes

### DISCUSSION THREAD

**Discussion by hertneky** [dtcenter/METplus#1351] [Configuration] [answered]
1. Regrid-Data-Plane
I am only needing to regrid a field from my fcst file, which works fine.

Since I don't need to regrid anything from OBS, I set OBS_REGRID_DATA_PLANE_RUN = False. To me, it seems that it should not try to run regridDataPlane for OBS with that setting, but it still does try and prints out errors in the logs. Am I misinterpreting what that config option is for?

2. Extract-Tiles 
Extract-Tiles is using the 2D MTD output for tracking the feature - for which there is a single 2D MTD output file for each initialization, which automatically includes the valid time of the 1st lead time in the file name. As I currently have the use-case set up, ExtractTiles looks for a MTD output file for each lead time in the LEAD_SEQ, since I have EXTRACT_TILES_MTD_INPUT_TEMPLATE = {init?fmt=%Y%m%d%H}/mtd_{valid?fmt=%Y%m%d_%H%M%S}V_2d.txt, which works but includes errors for all leads following the 1st one due to looping through the times, however it's the only way I have been able to run over all inits. I can't hard-code the valid time in the ExtractTiles template since it changes depending on the case init hour (00 or 12). I also couldn't figure out how to explicitly specify the filename of the MTD output. MTD_OUPUT_TEMPLATE did not seem to work on the filename - should it?

Note: These errors do not cause any issues with my run, but it would be nice to clean them up.

I am running on seneca (6 hr run for a single init):
log: /d1/personal/hertneky/metprd/logs/metplus.log.20220112080508
parm: /d1/personal/hertneky/parm/jtti_wpc_mtd_featureRelative.conf

**Comment by georgemccabe** [comment, marked as answer]
Hi @hertneky,

1 is happening due to a bug. The ExtractTiles wrapper creates an instance of RegridDataPlane to use to create the cutouts. That instance always runs RegridDataPlane on both OBS and FCST, so it overrides the FCST_REGRID_DATA_PLANE_RUN and OBS_REGRID_DATA_PLANE_RUN variables to do so. However, these changes to the config should be isolated to the instance of RegridDataPlane wrapper that is being used in ExtractTiles. Instead it is overriding those variables for the METplusConfig object used by all of the wrappers. No one has run both ExtractTiles and RegridDataPlane in the PROCESS_LIST before this to uncover this issue before. Thanks for bringing this to our attention! I created an issue to address it: #1356

For 2, you can't explicitly set the output file names for MTD output using the template. You can set the MTD_OUTPUT_PREFIX which will add the text set there to the output file paths, but the valid time will still be added.

I think currently there isn't a good way to set this up without having it process some times that you don't need to run. I have some proposed changes to the METplus time logic that would allow you to set the time looping settings differently for a given tool. This would allow you to only process the times you need for the tool you need. I created GitHub Discussion #1278 soliciting feedback about these changes. You could comment on this discussion noting that implementing these changes would benefit your use case to allow you to set the timing settings that you need.

**Reply by hertneky** [reply]
Thanks for the thorough replies George - very helpful. I will take a look at #1278. I need to think about it a bit first.
